climakitae/data_loaders.py

```diff
@@ -32,19 +33,22 @@ def _open_and_concat(file_list, selections, ds_region):
     all_files = xr.Dataset()
     for one_file in file_list:
         data = cat[one_file].to_dask()
+        attributes = data.attrs
         source_id = data.attrs["source_id"]
```

Comment:
Many combinations of variable, scenario, and temporal resolution result in 
`KeyError: 'source_id'` 

It would take a very, very long time to make an exhaustive list. Looks like it comes down to some missing metadata?

Reply:
Thanks for catching that. It seems to be the second time through and every time thereafter. I need to make that line a *copy* of the contents of data.attrs (it was treating it like a reference), so if a previous iteration removes an attribute then that removal won't persist.